opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -0,0 +1,307 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Tuple, Type, Union
+
+from opentelemetry import metrics as metrics_api
+
+logger = logging.getLogger(__name__)
+
+
+class Metric(metrics_api.Metric):
+    """See `opentelemetry.metrics.Metric`."""
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        unit: str,
+        value_type: metrics_api.ValueType,
+        label_keys: Tuple[str, ...] = None,
+        enabled: bool = True,
+        monotonic: bool = False,
+    ):
+        self.name = name
+        self.description = description
+        self.unit = (unit,)
+        self.value_type = value_type
+        self.label_keys = label_keys
+        self.enabled = enabled
+        self.monotonic = monotonic
+        self.handles = {}
+
+    def get_handle(
+        self, label_values: Tuple[str, ...]
+    ) -> metrics_api.MetricHandle:
+        """See `opentelemetry.metrics.Metric.get_handle`."""
+        pass  # pylint: disable=unnecessary-pass
+
+    def remove_handle(self, label_values: Tuple[str, ...]) -> None:
+        """See `opentelemetry.metrics.Metric.remove_handle`."""
+        self.handles.pop(label_values, None)
+
+    def clear(self) -> None:
+        """See `opentelemetry.metrics.Metric.clear`."""
+        self.handles.clear()
+
+
+class Counter(Metric):
+    """See `opentelemetry.metrics.Counter`.
+
+    By default, counter values can only go up (monotonic). Negative inputs
+    will be discarded for monotonic counter metrics. Counter metrics that
+    have a monotonic option set to False allows negative inputs.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        unit: str,
+        value_type: metrics_api.ValueType,
+        label_keys: Tuple[str, ...] = None,
+        enabled: bool = True,
+        monotonic: bool = True,
+    ):
+        super(Counter, self).__init__(
+            name,
+            description,
+            unit,
+            value_type,
+            label_keys=label_keys,
+            enabled=enabled,
+            monotonic=monotonic,
+        )
+
+    def get_handle(
+        self, label_values: Tuple[str, ...]
+    ) -> metrics_api.CounterHandle:
+        """See `opentelemetry.metrics.FloatCounter.get_handle`."""
+        handle = self.handles.get(
+            label_values,
+            CounterHandle(self.value_type, self.enabled, self.monotonic),
```

Comment:
Performance: You are always constructing a new handle object that way, even when one is found (the second argument to get is evaluated before get is called).